Source/WebCore/page/Quirks.cpp

```diff
@@ -359,6 +365,24 @@ bool Quirks::shouldAvoidUsingIOS17UserAgentForFacebook() const
 #endif
 }
 
+void Quirks::updateStorageAccessUserAgentStringQuirks(HashMap<RegistrableDomain, String>&& userAgentStringQuirks)
+{
+    auto& quirks = updatableStorageAccessUserAgentStringQuirks();
+    quirks.clear();
+    for (auto&& [domain, userAgent] : userAgentStringQuirks)
+        quirks.add(WTFMove(domain), WTFMove(userAgent));
+}
+
+String Quirks::storageAccessUserAgentStringQuirkForDomain(const URL& url)
+{
+    const auto& quirks = updatableStorageAccessUserAgentStringQuirks();
```

Comment:
It seems a bit odd that these are static, and not scoped to the document, like the other quirks (more concretely, this means that the quirk won't respect the value of `needsQuirks()`, which can be toggled by the user via Web Inspector UI).

Is this intentional?